### PR TITLE
Removing JSX references to MaybeInternalLink

### PR DESCRIPTION
**Pull request by ashleemboyer**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/) 🔎
- [Asana task](https://app.asana.com/0/1202097197789424/1202482837900324/f) 🎟️

## 🗒️ What

- Removes the `MaybeInteralLink` component and updates references to it to use `next/link`

## 🤷 Why

- `MaybeInternalLink` was incorrectly rendering `next/link` for external links and `<a>` for internal ones, so some links to internal pages triggered full reloads
- `next/link` can handle both internal and external links

### Documentation

- [ ] Go to [/swingset/components/cardlink](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/swingset/components/cardlink)
  - [ ] Make sure the links at the top of the page work
  - [ ] Make sure the page is still accurate
- [ ] Go to [/swingset/components/inlinelink](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/swingset/components/inlinelink)
  - [ ] Make sure the links at the top of the page work
  - [ ] Make sure the page is still accurate
- [ ] Go to [/swingset/components/standalonelink](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/swingset/components/standalonelink)
  - [ ] Make sure the links at the top of the page work
  - [ ] Make sure the page is still accurate

### `CardLink`

- [ ] Home page:
  - [ ] Vault card
  - [ ] HashiConf Global card
- [ ] Product landing views ([/vault](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/vault), [/waypoint](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/waypoint))
  - [ ] Icon cards just under the hero
  - [ ] Collection cards
  - [ ] Tutorial cards
  - [ ] Other linked cards in the page, which mostly link to docs
- [ ] Root docs landing views ([/vault/docs](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/vault/docs), [/waypoint/docs](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/waypoint/docs))
  - [ ] Icon cards just under the hero
  - [ ] Tutorial & Collection cards towards the bottom of the page
  - [ ] Other linked cards in the page, which mostly link to docs
- [ ] Tutorials landing pages ([/vault/tutorials](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/vault/tutorials), [/waypoint/tutorials](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/waypoint/tutorials))
  - [ ] Collection cards
  - [ ] Tutorial cards
- [ ] Collection views (such as [/waypoint/tutorials/get-started-docker](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/waypoint/tutorials/get-started-docker))
  - [ ] Tutorial cards
- [ ] Tutorial views (such as [/waypoint/tutorials/get-started-docker/get-started-intro](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/waypoint/tutorials/get-started-docker/get-started-intro))
  - [ ] Back/Next/etc. linked cards at the bottom of the page
  - [ ] "This tutorial also appears in" cards
- [ ] Install views ([/vault/downloads](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/vault/downloads), [/waypoint/downloads](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/waypoint/downloads))
  - [ ] "Official releases" card
  
### `InlineLink`

- [ ] The "most recent version" link on versioned 404 pages (such as [/waypoint/docs/v0.7.x/not-a-real-page)](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/waypoint/docs/v0.7.x/not-a-real-page)
- [ ] The "View latest version" link in the version alert on docs pages (such as [/waypoint/docs/v0.7.x](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/waypoint/docs/v0.7.x))
- [ ] The 3 links in the Notes section at the bottom of the "Operating System " card in install views ([/vault/downloads](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/vault/downloads), [/waypoint/downloads](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/waypoint/downloads))
- [ ] Inline links in tutorial content
- [ ] Inline links in docs content

### `StandaloneLink`

- [ ] Install views ([/vault/downloads](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/vault/downloads), [/waypoint/downloads](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/waypoint/downloads)):
  - [ ] Sidebar "Waypoint Home" link
  - [ ] Sidecar "Learn more" link
  - [ ] Under "Operating System" card, each "Download" link and the "GitHub" changelog link
- [ ] Home page:
  - [ ] In the learn section, the "Start learning" link
  - [ ] At the bottom of the page, the "Open a support ticket", "Find your answer on the forum", and "Join our community" links
- [ ] On [/vault](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/vault):
  - [ ] The "Learn more about Vault features" link
  - [ ] The "CLI Quick Start" link
  - [ ] The "Developer Quick Start" link
- [ ] On [/waypoint](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/waypoint):
  - [ ] The "Learn more about Vault features" link
  - [ ] The "CLI Quick Start" link
- [ ] On [/vault/tutorials](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/vault/tutorials), the "Explore HCP Vault" link
- [ ] On [/waypoint/tutorials](https://dev-portal-git-ambremove-maybe-internal-link-hashicorp.vercel.app/waypoint/tutorials), the "Get started in just a few minutes" link

### CODE REVIEW

File: src/components/standalone-link/index.tsx [hunk 1]
```diff
@@ -19,21 +19,26 @@ const StandaloneLink = ({
...
-      rel={openInNewTab ? 'noreferrer noopener' : undefined}
-      target={openInNewTab ? '_blank' : '_self'}
-    >
-      {iconPosition === 'leading' && icon}
-      <span className={classNames(s.text, textClassName)}>{text}</span>
-      {iconPosition === 'trailing' && icon}
-    </MaybeInternalLink>
+    <Link href={href}>
+      {/* TODO double check usage of the `onClick` prop */}
+      {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
```
Comment (ashleemboyer) [inline src/components/standalone-link/index.tsx:28]: 🗒️ Note to self: still need to double check this

File: src/components/inline-link/index.tsx [hunk 2]
```diff
@@ -19,10 +21,29 @@ const InlineLink = ({
     className
   )
 
+  /**
+   * This case is possible for some of the extra <a> elements that are added by
+   * our `anchor-links` remark plugin.
+   *
+   * FUTURE TODOs
+   *   - update the plugin to no longer add empty <a> elements
+   *   - update this to throw an error
```
Comment (zchsh) [inline src/components/inline-link/index.tsx:30]: 🙌 + 💯

File: src/components/standalone-link/index.tsx (hunk 1 above)
Comment (zchsh) [reply src/components/standalone-link/index.tsx:28]: 💭 Is this something that feels blocking? (If not, the `TODO` comment here seems fine by me 👍)

**Review by zchsh — APPROVED** [review]
🙌 This looks great to me! 

Also seeing how this could spark the idea for a `next/link` wrapper similar to `MaybeInternalLink`, but without the switching logic 👍

File: src/components/standalone-link/index.tsx (hunk 1 above)
Comment (ashleemboyer) [reply src/components/standalone-link/index.tsx]: Yes, it was a merge blocker for me! Just finished investigating it and updated the comment to explain why the warning was happening and why it was being disabled in this case.

**Pull request merged**